backup/backup.go

```diff
@@ -20,6 +21,132 @@ import (
 	"go.uber.org/zap"
 )
 
+type BucketManifestWriter struct {
+	ts *tenant.Service
+	mc *meta.Client
+}
+
+func NewBucketManifestWriter(ts *tenant.Service, mc *meta.Client) BucketManifestWriter {
+	return BucketManifestWriter{
+		ts: ts,
+		mc: mc,
+	}
+}
+
+// WriteManifest writes a bucket manifest describing all of the buckets that exist in the database.
+// It is intended to be used to write to an HTTP response after appropriate measures have been taken
+// to ensure that the request is authorized.
+func (b BucketManifestWriter) WriteManifest(ctx context.Context, w io.Writer) error {
+	bkts, _, err := b.ts.FindBuckets(ctx, influxdb.BucketFilter{})
+	if err != nil {
+		return err
+	}
+
+	l := []influxdb.BucketMetadataManifest{}
+
+	for _, bkt := range bkts {
+		org, err := b.ts.OrganizationService.FindOrganizationByID(ctx, bkt.OrgID)
+		if err != nil {
+			return err
+		}
+
+		dbInfo := b.mc.Database(bkt.ID.String())
+
+		l = append(l, influxdb.BucketMetadataManifest{
+			OrganizationID:         bkt.OrgID,
+			OrganizationName:       org.Name,
+			BucketID:               bkt.ID,
+			BucketName:             bkt.Name,
+			DefaultRetentionPolicy: dbInfo.DefaultRetentionPolicy,
+			RetentionPolicies:      retentionPolicyToManifest(dbInfo.RetentionPolicies),
+		})
+	}
+
+	return json.NewEncoder(w).Encode(&l)
+}
+
+// retentionPolicyToManifest and the various similar functions that follow are for converting
+// from the structs in the meta package to the manifest structs
+func retentionPolicyToManifest(meta []meta.RetentionPolicyInfo) []influxdb.RetentionPolicyManifest {
+	r := []influxdb.RetentionPolicyManifest{}
```

Comment:
Could pre-set the capacity here via `make` but it's probably less important since buckets in 2.x only have 1 retention policy